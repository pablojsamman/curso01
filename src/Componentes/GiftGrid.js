import React from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

const GiftGrid = ({categoria}) => {
    
    const {data, loading} = useFetchGifs(categoria)
    
    return (
    <>
        <h3> {categoria} </h3>
        {loading&&<p>Cargando...</p>}


        <div className='card-grid'>

            {
                data.map( img => {
                    return <GifGridItem key={img.id} { ...img }  />
                })
            }
        </div>
    </>
    )
}

GiftGrid.propTypes = {
    categoria: PropTypes.string.isRequired,
}

export default GiftGrid
