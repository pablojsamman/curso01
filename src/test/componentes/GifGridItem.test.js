import React from 'react'
import {shallow} from 'enzyme'
import { GifGridItem } from '../../Componentes/GifGridItem'


describe('Pruebas en <GifGridItem>', () => {
    const title = 'Algun titulo'
    const url = 'alguna url'
    const wrapper = shallow( <GifGridItem title={title} url={url} /> )

    test('debe mostrar componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('debe de tener un parrafo con el titulo', () => {
        
        const p = wrapper.find('p')
        expect( p.text().trim() ).toBe(title)

    })

    test('debe de tener la imagen igual al url y alt', () => {
        
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)

    })

    
    
})