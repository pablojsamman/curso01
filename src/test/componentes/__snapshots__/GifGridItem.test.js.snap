// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem> debe mostrar componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Algun titulo"
    src="alguna url"
  />
  <p>
     
    Algun titulo
     
  </p>
</div>
`;
